### DISCUSSION TITLE
Context menu icons do not take effect.

### DISCUSSION THREAD

**Discussion by un4gt** [eclipse-theia/theia#12954] [Q&A] [answered]
I added additional context menus to theia.
```typescript
export namespace SpeakCommands {
    export const SPEAK_HELLO = Command.toLocalizedCommand({
        id: 'navigator.speak.hello',
        label: 'Speak Hello',
        iconClass: 'fa fa-star-o'
    }, 'theia/navigator/speakHello');

menus.registerMenuAction([...NAVIGATOR_CONTEXT_MENU, 'speak_hello'], {
            commandId: SpeakCommands.SPEAK_HELLO.id,
            label: SpeakCommands.SPEAK_HELLO.label,
            icon: 'fa fa-star-o'
        });
```
![image](https://github.com/eclipse-theia/theia/assets/81068011/ccfe0bd8-9527-4202-96aa-8992c1161f87)
*I've unregistered the other menus.*
The menu works fine, but I noticed that the icon I set didn't work.
![image](https://github.com/eclipse-theia/theia/assets/81068011/059d0564-b479-4f77-9d19-c13d6cc350c7)
Is it possible that the context menu has special requirements for icons or that I'm not setting the icon in the right way?

**Comment by msujew** [comment, marked as answer]
There seems to be an issue in our [menus.css](https://github.com/eclipse-theia/theia/blob/bc712cadffc8994c984729d2ab2773cc9a12fbaa/packages/core/src/browser/style/menus.css#L208-L221) that only allows to show codicons there right now. The Icon is correctly set, but the font is incorrectly set. You can probably create a style override for your item to fix this in your own extension.

**Reply by un4gt** [reply]
Thank you!@msujew .
```css
.p-Menu-itemIcon::before,
.p-Menu-itemSubmenuIcon::before {
    font: inherit;
}
```
I was able to get the icon to display properly this way
